apps/clock/js/ring_view.js

```diff
@@ -137,8 +137,25 @@ RingView.prototype = {
     // Display the proper screen widgets.
     this.ringDisplay.dataset.ringType = alert.type;
 
+    var displayedTime;
+    if (alert.type === 'alarm') {
+      // Display the time as specified with "hour" and "minute". The timestamp
+      // stored by mozAlarms does not move when the timezone changes, so we
+      // cannot just use the value stored in '.time'. See
+      // https://bugzilla.mozilla.org/show_bug.cgi?id=1178208#c4
+      displayedTime = new Date();
+      displayedTime.setHours(alert.hour);
+      displayedTime.setMinutes(alert.minute);
```

Comment:
The alarm UI might trigger long after the "scheduled alarm time" has passed, due to the phone being off or whatever. For UX reasons we display the time the alarm was supposed to go off, not the time we instantiated the dialog.